### DISCUSSION TITLE
Tracking visitors of my Birdnet-Pi station

### DISCUSSION THREAD

**Discussion by blogdemoi** [mcguirepr89/BirdNET-Pi#510] [Q&A] [answered]
Hi,

I wish to know the origins of my eventual visitors of my Birdnet-Pi.

For that I will install Umami https://umami.is/ on my server. I need to insert code in the head section of my generated Birdnet-Pi website.

Could you tell me where is the page, where to insert that code?

Thank you.

**Comment by ehpersonal38** [comment, marked as answer]
You can insert any header code in `/home/pi/BirdNET-Pi/homepage/views.php`. There may be a catch in that you won't be able to see if users visit specific pages, since something like /todays_detections.php actually has the HTTP request of /views.php?view=Today's Detections (so technically the same file name), but maybe that's something your software will deal with. Just a heads up, in case you come across that scenario. :)

**Reply by blogdemoi** [reply]
Thank you.

**Comment by Svardsten53** [comment]
I use Matomo (formerly Piwik) for visitors tracking and have inserted the code in views.php before the end body tag.

**Reply by blogdemoi** [reply]
Thank you.

**Reply by blogdemoi** [reply]
Hi,

I installed Matomo on my server and inserted the following code in views.php:

```
</script>
</div>
<!-- Matomo -->
<script>
  var _paq = window._paq = window._paq || [];
  /* tracker methods like "setCustomDimension" should be called before "trackPa>
  _paq.push(['trackPageView']);
  _paq.push(['enableLinkTracking']);
  (function() {
    var u="//arnauld.org/matomo/";
    _paq.push(['setTrackerUrl', u+'matomo.php']);
    _paq.push(['setSiteId', '2']);
    var d=document, g=d.createElement('script'), s=d.getElementsByTagName('scri>
    g.async=true; g.src=u+'matomo.js'; s.parentNode.insertBefore(g,s);
  })();
</script>
<!-- End Matomo Code -->
</body>
```

However I don't have any visits even when I am "self" visiting.
Could you tell me if you did the same way or another way?

Thank you.

**Reply by Svardsten53** [reply]
I use the following code and it works. Two of your lines seem to be truncated.

```<!-- Matomo -->
<script type="text/javascript">
  var _paq = window._paq = window._paq || [];
  /* tracker methods like "setCustomDimension" should be called before "trackPageView" */
  _paq.push(['trackPageView']);
  _paq.push(['enableLinkTracking']);
  (function() {
    var u="https://piwik.svardsten.com/";
    _paq.push(['setTrackerUrl', u+'matomo.php']);
    _paq.push(['setSiteId', '11']);
    var d=document, g=d.createElement('script'), s=d.getElementsByTagName('script')[0];
    g.type='text/javascript'; g.async=true; g.src=u+'matomo.js'; s.parentNode.insertBefore(g,s);
  })();
</script>
<!-- End Matomo Code -->
```

**Reply by blogdemoi** [reply]
Thank you. I am just back home and noticed (I didn't make any change in views.php yet) I had one visitor from Cologne at 18:33:59. So I will wait a few days before changing anything... What intrigues me is that my own views don't appear.

Edit: Hum, I just realized that I didn't copy/paste correctly in my previous post and the two lines you mentioned aren't truncated in my views.php...

**Reply by mcguirepr89** [reply]
I just wanted to put this out there as my due diligence: [GDPR](https://gdpr.eu/)

**Reply by blogdemoi** [reply]
> I just wanted to put this out there as my due diligence: [GDPR](https://gdpr.eu/)

Thank you. What am I supposed to do with this? Do you think I shouldn't track my visitors? Or maybe I should add a warning that visitors are being tracked? In that case I will see in the next few days how I can implement this :-)

**Reply by Svardsten53** [reply]
Not all visitors are shown. There may be browser privacy protections, virus protections or other security features enabled that prevent any responses from being sent to the Matomo script.

**Reply by mcguirepr89** [reply]
@blogdemoi 
> What am I supposed to do with this? 

Read all of it before collecting data from citizens of the EU without their consent.

I am no authority and this should not be considered legal advise, but I think you'll have to allow visitors who are citizens of the EU an opportunity to opt out of you collecting their IP address. More specifically, you'll need visitors to allow that permission _before_ you collect the data. I would be _extremely_ careful with this data collection.

**Reply by blogdemoi** [reply]
> Read all of it before collecting data from citizens of the EU without their consent.
> 
> I am no authority and this should not be considered legal advise, but I think you'll have to allow visitors who are citizens of the EU an opportunity to opt out of you collecting their IP address. More specifically, you'll need visitors to allow that permission before you collect the data. I would be extremely careful with this data collection.

Thank you. You are right, I will see how I can implement such a thing in the next few days. I guess it can be easily made with Matomo, let's read the documentation...

**Reply by Svardsten53** [reply]
Matomo and GDPR are clarified here:
https://matomo.org/gdpr-analytics/

**Reply by mcguirepr89** [reply]
Thank you, @Svardsten53!! That is perfect

**Comment by blogdemoi** [comment]
Finally I decided not to use Matomo, I prefer not to track users, thinking about this I decided to get rid of Matomo in my Hugo blog too :-)
